components/Accordion/src/AccordionToggle.js

```diff
@@ -1,59 +1,25 @@
-const AccordionToggle = (className = 'denhaag-accordion__title') => {
-  // Select all clickable elements.
-  [...document.getElementsByClassName(className)]?.forEach((element) => {
-    const wrapper = element.closest('.denhaag-accordion__container');
-    let accordionHeight = `${wrapper.querySelector('.denhaag-accordion__details-content').offsetHeight}px`;
-
-    const setAccordionHeight = (el, newHeight) => {
-      el.style.setProperty('--denhaag-accordion-details-max-height', newHeight);
-    };
-
-    const setAccordionDisplay = (el, value) => {
-      const content = el.querySelector('.denhaag-accordion__details');
-      content.style.display = value;
-    };
-
-    // Set max-height property.
-    if (wrapper.querySelector('.denhaag-accordion__details-content')) {
-      setAccordionHeight(wrapper, accordionHeight);
-    }
-
-    // Toggle accordion-item.
-    element.onclick = () => {
-      const expanded = element.getAttribute('aria-expanded') === 'false' ? 'true' : 'false';
-
-      if (expanded === 'false') {
-        // Return correct height to show closing animation
-        setAccordionHeight(wrapper, accordionHeight);
-
-        setTimeout(() => {
-          wrapper.classList.toggle('denhaag-accordion__container--open');
-        }, 50);
-
-        setTimeout(() => {
-          setAccordionDisplay(wrapper, 'none');
-        }, 500);
-      }
-
-      if (expanded === 'true') {
-        setAccordionDisplay(wrapper, 'block');
-        setTimeout(() => {
-          const dynamicHeight = `${wrapper.querySelector('.denhaag-accordion__details-content').offsetHeight}px`;
-          accordionHeight = dynamicHeight;
-          setAccordionHeight(wrapper, dynamicHeight);
-          wrapper.classList.toggle('denhaag-accordion__container--open');
-        }, 50);
-
-        // Wait 500 ms (transition duration) to set height to auto, to ensure the animation works
-        setTimeout(() => {
-          setAccordionHeight(wrapper, 'auto');
-        }, 550);
-      }
-
-      // Set opposite aria-expanded.
-      element.setAttribute('aria-expanded', expanded);
-    };
-  });
-};
-
-export default AccordionToggle;
+export default class Accordion {
+  constructor(className = 'denhaag-accordion__container') {
+    this.collapses = document.getElementsByClassName(className);
```

Comment:
constructor is called before DOMContentLoaded -> so, there's a possibility getElementsByClassName returns none.

Might want to add an `initialize()` method?

Reply:
Discussed, can be fired on constructor since class is that tiny